#! /bin/sh -e
## 16_update_debian.dpatch by Tollef Fog Heen <tfheen@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: update is called from the Debian maintainer scripts and prints
## DP: non-appropriate output.  Change that.

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $patch_opts -p1 < $0;;
    -unpatch) patch $patch_opts -p1 -R < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0
@DPATCH@
--- mailman-2.1.4.orig/bin/update
+++ mailman-2.1.4/bin/update
@@ -295,8 +295,7 @@
         else:
             # directory
             print _("""\
-    looks like you have a really recent CVS installation...
-    you're either one brave soul, or you already ran me""")
+    Your installation seems up-to-date, great!""")
 
 
     #
@@ -318,8 +317,7 @@
         %(newname)s""")
         else: # directory
             print _("""\
-    looks like you have a really recent CVS installation...
-    you're either one brave soul, or you already ran me""")
+    Your installation seems up-to-date, great!""")
 
     #
     # move the html archives there
@@ -333,31 +331,6 @@
     # BAW: Is this still necessary?!
     mlist.Save()
     #
-    # check to see if pre-b4 list-specific templates are around
-    # and move them to the new place if there's not already
-    # a new one there
-    #
-    tmpl_dir = os.path.join(mm_cfg.PREFIX, "templates")
-    list_dir = os.path.join(mm_cfg.PREFIX, "lists")
-    b4_tmpl_dir = os.path.join(tmpl_dir, mlist._internal_name)
-    new_tmpl_dir = os.path.join(list_dir, mlist._internal_name)
-    if os.path.exists(b4_tmpl_dir):
-        print _("""\
-- This list looks like it might have <= b4 list templates around""")
-        for f in os.listdir(b4_tmpl_dir):
-            o_tmpl = os.path.join(b4_tmpl_dir, f)
-            n_tmpl = os.path.join(new_tmpl_dir, f)
-            if os.path.exists(o_tmpl):
-                if not os.path.exists(n_tmpl):
-                    os.rename(o_tmpl, n_tmpl)
-                    print _('- moved %(o_tmpl)s to %(n_tmpl)s')
-                else:
-                    print _("""\
-- both %(o_tmpl)s and %(n_tmpl)s exist, leaving untouched""")
-            else:
-                print _("""\
-- %(o_tmpl)s doesn't exist, leaving untouched""")
-    #
     # Move all the templates to the en language subdirectory as required for
     # Mailman 2.1
     #
@@ -428,12 +401,12 @@
                 'Mailman/HyperDatabase.py', 'Mailman/pipermail.py',
                 'Mailman/smtplib.py', 'Mailman/Cookie.py',
                 'bin/update_to_10b6', 'scripts/mailcmd',
-                'scripts/mailowner', 'mail/wrapper', 'Mailman/pythonlib',
+                'scripts/mailowner', 'Mailman/pythonlib',
                 'cgi-bin/archives', 'Mailman/MailCommandHandler'):
         remove_old_sources(mod)
     listnames = Utils.list_names()
     if not listnames:
-        print _('no lists == nothing to do, exiting')
+#        print _('no lists == nothing to do, exiting')
         return
     #
     # for people with web archiving, make sure the directories
